src/overlays/actors/ovl_En_Snowman/z_en_snowman.c

```diff
@@ -42,155 +73,1123 @@ const ActorInit En_Snowman_InitVars = {
+    this->collider.base.acFlags &= ~(AC_ON | AC_HIT);
+    this->collider.base.ocFlags1 &= ~(OC1_ON | OC1_HIT);
+    this->actionFunc = EnSnowman_MarkForDeath;
+}
+
+/**
+ * Simply calls Actor_MarkForDeath on the Eeno.
+ */
+void EnSnowman_MarkForDeath(EnSnowman* this, GlobalContext* globalCtx) {
+    Actor_MarkForDeath(&this->actor);
+}
+
+/**
+ * Creates a small split Eeno with a certain Y-rotation and at a certain offset from basePos.
+ */
+void EnSnowman_CreateSplitEeno(EnSnowman* this, Vec3f* basePos, s32 yRot) {
+    this->actor.flags |= ACTOR_FLAG_1;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.shape.rot.y = yRot;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->combineState = EN_SNOWMAN_COMBINE_STATE_NONE;
+    this->actor.colChkInfo.health = 2;
+    this->eenoScale = 1.0f;
+    this->actor.world.pos.x = (Math_SinS(yRot) * 40.0f) + basePos->x;
+    this->actor.world.pos.y = basePos->y;
+    this->actor.world.pos.z = (Math_CosS(yRot) * 40.0f) + basePos->z;
+    this->combineTimer = 600;
+    this->actor.params = EN_SNOWMAN_TYPE_SPLIT;
+    this->actor.flags &= ~ACTOR_FLAG_400;
+    this->collider.base.ocFlags1 |= OC1_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    EnSnowman_SetupMoveSnowPile(this);
+}
+
+/**
+ * Absorbs the smaller Eeno into the larger one. The smaller Eeno's target scale is set to
+ * zero so it gradually disappears, and its collision is disabled, while the larger Eeno's
+ * target scale increases.
+ */
+void EnSnowman_AbsorbEeno(EnSnowman* smallerEeno, EnSnowman* largerEeno) {
+    Actor_PlaySfxAtPos(&largerEeno->actor, NA_SE_EN_YMAJIN_UNITE);
+    largerEeno->work.targetScaleDuringCombine += 0.005f;
+    smallerEeno->combineState = EN_SNOWMAN_COMBINE_STATE_BEING_ABSORBED_OR_DONE;
+    smallerEeno->collider.base.ocFlags1 &= ~OC1_HIT;
+    smallerEeno->collider.base.acFlags &= ~AC_HIT;
+    smallerEeno->collider.base.ocFlags1 &= ~OC1_ON;
+    smallerEeno->collider.base.acFlags &= ~AC_ON;
+    smallerEeno->work.targetScaleDuringCombine = 0.0f;
+}
+
+void EnSnowman_SetupCombine(EnSnowman* this, GlobalContext* globalCtx, Vec3f* combinePos) {
+    if (this->actor.colChkInfo.health == 0) {
+        this->combineState = EN_SNOWMAN_COMBINE_STATE_NO_ABSORPTION;
+    } else {
+        this->actor.flags |= ACTOR_FLAG_10;
+        Math_Vec3f_Copy(&this->combinePos, combinePos);
+        this->combineState = EN_SNOWMAN_COMBINE_STATE_ACTIVE;
+
+        if (this->actionFunc != EnSnowman_Submerge) {
+            this->combineTimer = 400;
+
+            //! @bug: Skipping this based the current actionFunc results in whatever is currently in work
+            // being treated as the target scale when this function is called from EnSnowman_Submerge.
+            // Since the only other things that get put in work are animation frame numbers, this results
+            // in an enormous target scale, potentially creating the so-called "Mega Eeno" glitch.
```

Comment:
No, because of how EnSnowman_Combine works. If two Eenos have the same scale, then the one that is currently executing EnSnowman_Combine absorbs the other one. The bug gives the Eeno a massive target scale, but it has to call EnSnowman_Combine to start changing its actual scale, and it grows rather slowly. So even if an Eeno is setup to become the Mega Eeno, it could be absorbed before it even gets a chance to run EnSnowman_Combine.

Reply:
Hmm, that needs to be explained a bit more explicitly somewhere, I think. Also, when you do a `@bug` comment, you have to do `//!` on every line or Doxygen ignores it, it turns out.

Reply:
OK, added a sentence to this comment with that info. Let me know if it can be improved.